### ISSUE TITLE
200 not possible for OPTIONS with empty response content

### ISSUE THREAD

**Issue by paq85** [FriendsOfSymfony/FOSRestBundle#1126]
Hi,

AFAIK OPTIONS request should return 200 with no content, just header with "Allowed" for example.
http://www.w3.org/Protocols/rfc2616/rfc2616-sec9.html

It's not possible to do it using FOSRestBundle

Eg. following code would still result in 204 (No content) instead of 200
```php
        $headers = ['Allow' => 'GET,POST,DELETE'];
        return $this->handleView($this->view(null, Response::HTTP_OK, $headers));
```

It's because `vendor/friendsofsymfony/rest-bundle/FOS/RestBundle/View/ViewHandler.php` has following check
```php
        if (200 !== ($code = $view->getStatusCode())) {
            return $code;
        }

        return null !== $content ? Codes::HTTP_OK : $this->emptyContentCode;
```
which overrides my requested 200 to 204 (by default)

Please consider making it possible to send 200 with no content.

**Comment by lsmith77** [comment]
hmm yeah .. we might be overdoing the "magic" here .. one way to solve this would be to check the http verb and limit this behavior to POST/PUT .. 

maybe the cleaner thing would be to stop defaulting to a specific status code and leaving it at ``null`` so that we know if an explicit status code is set, we should always return that, rather than doing any sort of magic. though this would be a change for 2.x only.

**Comment by JHGitty** [comment]
Fixed and merged in #1299. @paq85 @lsmith77
